geostore/s3_utils.py

```diff
@@ -52,3 +53,43 @@ def s3_url_reader(staging_url: str) -> GeostoreS3Response:
     staging_s3_client = get_s3_client_for_role(s3_role_arn)
     geostore_s3_client = get_s3_client_for_role(get_param(ParameterName.S3_USERS_ROLE_ARN))
     return s3_url_reader
+
+
+def get_s3_etag(s3_bucket: str, s3_object_key: str, logger: Logger) -> Optional[str]:
+    geostore_s3_client = get_s3_client_for_role(get_param(ParameterName.S3_USERS_ROLE_ARN))
+
+    try:
+        s3_response = geostore_s3_client.head_object(Bucket=s3_bucket, Key=s3_object_key)
+        return s3_response["ETag"]
+    except ClientError as error:
+        if error.response["Error"]["Code"] != "404":
+            logger.debug(
+                f"Unable to fetch eTag for “{s3_object_key}” in s3://{s3_bucket} due to “{error}”",
+                extra={GIT_COMMIT: get_param(ParameterName.GIT_COMMIT)},
+            )
+        # rather than raise, we return an empty string, indicating that the etag is different
+        # thus allowing the next step to continue rather than stalling the entire process
+        return None
+
+
+def calculate_s3_etag(body: bytes) -> str:
+    # https://awscli.amazonaws.com/v2/documentation/api/latest/topic/s3-config.html#multipart-chunksize
+    s3_default_chunk_size = 8_388_608  # Default value is 8 * 1024 * 1024
+
+    if body == b"":
+        return f'"{hashlib.md5().hexdigest()}"'
```

Comment:
Since this value is static and equal to the value in `should_return_accurate_etag_value_for_empty_file` you could just pull out a constant for it. In that case the `assert local_etag == expected_empty_file_etag` might be redundant. What do you think?